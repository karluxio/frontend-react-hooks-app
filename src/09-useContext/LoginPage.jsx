import { useContext } from "react"
import { UserContext } from "./context/UserContext";

export const LoginPage = () => {

  const { user, setUser } = useContext(UserContext)

  return (
    <>
      <h1>Login Page</h1>
      <hr />

      <pre aria-label="pre">
        {JSON.stringify(user, null, 2)}
      </pre>

      <button onClick={() => setUser({ id: 123, name: 'Luciano', email: 'luc@email.com' })} className="btn btn-primary">login</button>
    </>
  )
}