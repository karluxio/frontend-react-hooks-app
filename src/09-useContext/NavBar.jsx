import { NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
    <>
      <nav className="navbar navbar-expand-lg navbar-dark bg-dark p-3 rounded-3">
        <NavLink className="navbar-brand" to="/">UseContext</NavLink>

        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className="nav-item">
              <NavLink className="nav-link" to="/about">About</NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link" to="/login">Login</NavLink>
            </li>
          </ul>
        </div>
      </nav>
    </>
  )
}