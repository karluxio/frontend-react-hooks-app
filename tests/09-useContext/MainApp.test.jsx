import { render, screen } from '@testing-library/react'
import { MainApp } from '../../src/09-useContext/MainApp'
import { MemoryRouter } from 'react-router-dom'

describe("MainApp", () => {
  it("should match against snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <MainApp />
      </MemoryRouter>
    )

    expect(container).toMatchSnapshot()
  })

  it("should render Home page", () => {
    render(
      <MemoryRouter>
        <MainApp />
      </MemoryRouter>
    )
    // screen.debug()
    expect(screen.getByRole('heading', { level: 1, name: /home/i }))
  })

  it("should render Login page", () => {
    render(
      <MemoryRouter initialEntries={['/login']}>
        <MainApp />
      </MemoryRouter>
    )
    // screen.debug()
    expect(screen.getByRole('heading', { level: 1, name: /login/i }))
    expect(screen.getByRole("link", { name: /login/i }).className).toContain("active")
  })

  it("should render About page", () => {
    render(
      <MemoryRouter initialEntries={['/about']}>
        <MainApp />
      </MemoryRouter>
    )
    // screen.debug()
    expect(screen.getByRole('heading', { level: 1, name: /about/i }))
    expect(screen.getByRole("link", { name: /about/i }).className).toContain("active")
  })

  it("should render Home page in /*", () => {
    render(
      <MemoryRouter initialEntries={['/any-other-path']}>
        <MainApp />
      </MemoryRouter>
    )
    // screen.debug()
    expect(screen.getByRole('heading', { level: 1, name: /home/i })).toBeTruthy()
  })
})