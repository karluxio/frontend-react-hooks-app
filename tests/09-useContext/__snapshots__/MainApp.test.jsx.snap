// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp should match against snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark p-3 rounded-3"
  >
    <a
      aria-current="page"
      class="navbar-brand active"
      href="/"
    >
      UseContext
    </a>
    <div
      class="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        class="navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <hr />
  <h1>
    Home page - 
  </h1>
  <hr />
  <pre
    aria-label="pre"
  />
</div>
`;
